### PR TITLE
Companion #10403: Remove Default for AccountId

### PR DESCRIPTION
**Pull request by gavofyork**
paritytech/substrate#10403

Cumulus companion: paritytech/cumulus#842

### CODE REVIEW

**Comment by drahnr** [comment]
Some of these changes are overlapping with #4484 which will remove the `Default` implementations and introduce a helper crate to isolate the `fn dummy_*` helpers.

File: Cargo.lock [hunk 1]
```diff
@@ -10985,7 +10985,7 @@ version = "1.6.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "1f559b464de2e2bdabcac6a210d12e9b5a5973c251e102c44c585c71d51bd78e"
 dependencies = [
- "cfg-if 1.0.0",
+ "cfg-if 0.1.10",
```
Comment (gavofyork) [inline Cargo.lock:10988]: Don't know what this is about; I didn't do anything particular outside of using `diener`.

Comment (ordian) [reply Cargo.lock:10988]: a known bug in cargo:
* https://github.com/rust-lang/cargo/issues/5702
* https://github.com/rust-lang/cargo/issues/5529

Comment (gavofyork) [reply Cargo.lock:10988]:
```suggestion
 "cfg-if 1.0.0",
```

**Review by shawntabrizi — APPROVED** [review]
lgtm

**Pull request merged**